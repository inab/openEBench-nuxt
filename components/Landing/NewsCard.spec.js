import { mount } from '@vue/test-utils';
import NewsCard from './NewsCard.vue';

const factory = (propsData) => {
	return mount(NewsCard, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
		propsData,
	});
};

const propsData = {
	img: 'path',
	title: 'News Title',
	desc: 'News Description',
};

describe('NewsCard', () => {
	it('is instantiated', () => {
		const wrapper = factory(propsData);
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory(propsData);

		expect(wrapper.html()).toMatchSnapshot();
	});
});
