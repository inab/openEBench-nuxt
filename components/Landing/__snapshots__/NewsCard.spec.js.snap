// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard should match snapshot 1`] = `
"<div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 400px;\\">
  <div class=\\"v-image v-responsive white--text theme--light\\" style=\\"height: 200px;\\">
    <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
      <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
    </transition-stub>
    <div class=\\"v-responsive__content\\"></div>
  </div>
  <div class=\\"v-card__title\\">
    News Title
  </div>
  <div class=\\"v-card__text text-left\\"><span class=\\"text--primary\\">
			News Description
		</span></div>
  <div class=\\"v-card__actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Read more </span></button></div>
</div>"
`;
